Tasks/GoTool/gotool.ts

```diff
@@ -48,24 +48,26 @@ async function acquireGo(version: string): Promise<string> {
     //
     // Download - a tool installer intimately knows how to get the tool (and construct urls)
     //
-    version = toolLib.cleanVersion(version);
     let fileName: string = getFileName(version);
     let downloadUrl: string = getDownloadUrl(fileName);
     let downloadPath: string = null;
     try {
         downloadPath = await toolLib.downloadTool(downloadUrl);
     } catch (error) {
         tl.debug(error);
-        throw (tl.loc("VersionNotSupported", version));
+        throw (tl.loc("FailedToDownload", version, error));
     }
 
+    //make sure agent version is latest then 2.115.0
+    tl.assertAgent('2.115.0');
+
     //
     // Extract
     //
     let extPath: string;
     extPath = tl.getVariable('Agent.TempDirectory');
     if (!extPath) {
-        throw new Error('Expected Agent.TempDirectory to be set');
+        throw new Error(tl.loc("TempDirNotSet"));
```

Comment:
instead of throwing, should we set it to some default path

Reply:
NotFixing: making it consistent with other task such as node installer and DotNetCoreInstaller